### PR TITLE
kubectl-gadget: 0.14.0 -> 0.15.0

### PR DESCRIPTION
**Pull request by kranurag7**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Zaedus** [comment]
Result of `nixpkgs-review pr 224819` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>kubectl-gadget</li>
  </ul>
</details>

**Review by emilylange — CHANGES_REQUESTED** [review]
Please change your commit message to `kubectl-gadget: 0.14.0 -> 0.15.0` to be on par with the format outlined in https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md :)

Changelog: https://github.com/inspektor-gadget/inspektor-gadget/releases/tag/v0.15.0

Also, the package builds, but the result is empty:
```bash
❯ nix-build -A kubectl-gadget
/nix/store/c9jyvl0wnlxgbq51s0djn64c735kiinx-kubectl-gadget-0.15.0

❯ tree /nix/store/c9jyvl0wnlxgbq51s0djn64c735kiinx-kubectl-gadget-0.15.0
/nix/store/c9jyvl0wnlxgbq51s0djn64c735kiinx-kubectl-gadget-0.15.0

0 directories, 0 files
```

---

Result of `nixpkgs-review pr 224819` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>kubectl-gadget</li>
  </ul>
</details>

**Comment by kranurag7** [comment]
> the package builds, but the result is empty:

Hey @IndeedNotJames I'm not sure what I'm missing here. I checked it locally as well and the binary is missing. In the first attempt, I was only looking for successful builds and once it was successful, I raised a PR. Can you please guide me here or give me a reference for this? 

I tried the same with previous version and that seems to be working. 
```bash
$ tree /nix/store/l2wd1jsm86x4c1g8sr5nmk8i172c6ryf-kubectl-gadget-0.14.0
/nix/store/l2wd1jsm86x4c1g8sr5nmk8i172c6ryf-kubectl-gadget-0.14.0
└── bin
    └── kubectl-gadget

1 directory, 1 file
```

**Comment by IndeedNotJames** [comment]
Seems to require CGO now.
Note the line
```
imports github.com/seccomp/libseccomp-golang: build constraints exclude all Go files in /build/source/vendor/github.com/seccomp/libseccomp-golang
```
when building with `CGO_ENABLED = 0;`:

```
this derivation will be built:
  /nix/store/jmscgs1ccpvcy4fxpgh308annc09lfvz-kubectl-gadget-0.15.0.drv
building '/nix/store/jmscgs1ccpvcy4fxpgh308annc09lfvz-kubectl-gadget-0.15.0.drv'...
unpacking sources
unpacking source archive /nix/store/4fyaqmfzvq5y5ypzzixfi2r5iwjpk8sn-source
source root is source
patching sources
configuring
building
Building subPackage ././cmd/kubectl-gadget
package github.com/inspektor-gadget/inspektor-gadget/cmd/kubectl-gadget
        imports github.com/inspektor-gadget/inspektor-gadget/pkg/all-gadgets
        imports github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets/trace/capabilities/tracer
        imports github.com/seccomp/libseccomp-golang: build constraints exclude all Go files in /build/source/vendor/github.com/seccomp/libseccomp-golang
installing
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/7hpdyhdrpphlk69493mw2f5qrb8c1z8f-kubectl-gadget-0.15.0
checking for references to /build/ in /nix/store/7hpdyhdrpphlk69493mw2f5qrb8c1z8f-kubectl-gadget-0.15.0...
patching script interpreter paths in /nix/store/7hpdyhdrpphlk69493mw2f5qrb8c1z8f-kubectl-gadget-0.15.0
```

Use the following diff instead:
```diff
diff --git a/pkgs/applications/networking/cluster/kubectl-gadget/default.nix b/pkgs/applications/networking/cluster/kubectl-gadget/default.nix
index 1ef31d1f118..db5ec0ba46a 100644
--- a/pkgs/applications/networking/cluster/kubectl-gadget/default.nix
+++ b/pkgs/applications/networking/cluster/kubectl-gadget/default.nix
@@ -1,4 +1,4 @@
-{ lib, buildGoModule, fetchFromGitHub }:
+{ lib, buildGoModule, fetchFromGitHub, pkg-config, libseccomp }:
 
 buildGoModule rec {
   pname = "kubectl-gadget";
@@ -13,7 +13,9 @@ buildGoModule rec {
 
   vendorHash = "sha256-Y76Y3KR80dCx8+f6M0h5J6glGQGhXYE2KQM4jdJcDEM=";
 
-  CGO_ENABLED = 0;
+  CGO_ENABLED = 1;
+  nativeBuildInputs = [ pkg-config ];
+  buildInputs = [ libseccomp ];
 
   ldflags = [
     "-X main.version=v${version}"
```

```bash
❯ nix-build -A kubectl-gadget
/nix/store/hqv1f84bvi0gi1n13rmmf1api3d8d7j9-kubectl-gadget-0.15.0

❯ tree /nix/store/hqv1f84bvi0gi1n13rmmf1api3d8d7j9-kubectl-gadget-0.15.0
/nix/store/hqv1f84bvi0gi1n13rmmf1api3d8d7j9-kubectl-gadget-0.15.0
└── bin
    └── kubectl-gadget

1 directory, 1 file

❯ /nix/store/hqv1f84bvi0gi1n13rmmf1api3d8d7j9-kubectl-gadget-0.15.0/bin/kubectl-gadget version
Client version: v0.15.0
[...]
```

**Comment by IndeedNotJames** [comment]
Oh right, upstream uses `-extldflags '-static'`[^1]  and `withoutebpf`[^2] with `CGO_ENABLED=0`[^3].
Guess this works as well.
Nice catch.

But please don't forget to squash and reword your commit :) 


[^1]: https://github.com/inspektor-gadget/inspektor-gadget/blob/541f22ef10294aef62d19855cf8e8855bc45aecb/Makefile#L51
[^2]: https://github.com/inspektor-gadget/inspektor-gadget/blob/541f22ef10294aef62d19855cf8e8855bc45aecb/Makefile#L119
[^3]: https://github.com/inspektor-gadget/inspektor-gadget/blob/541f22ef10294aef62d19855cf8e8855bc45aecb/Makefile#L116

**Review by emilylange — APPROVED** [review]

**Comment by NickCao** [comment]
CGO is enabled by buildGoModule by default. I think `-extldflags '-static'` might not be actually needed.

**Comment by kranurag7** [comment]
> I think `-extldflags '-static'` might not be actually needed.

I have packaged this referencing the upstream project. Please, let me know if any changes required.

**Comment by NickCao** [comment]
Not strictly a request for change, just that we should be aware of the implications of building with CGO disabled/statically: NSS modules no longer work and DNS resolution can behave funny. So if it build and runs as a dynamically linked binary (linking to glibc for the dns resolution functions), I would prefer that.

File: pkgs/applications/networking/cluster/kubectl-gadget/default.nix [hunk 1]
```diff
@@ -2,21 +2,26 @@
...
+    hash = "sha256:1gn09kpkw8q2lxc8nic7hd1lhp0z4vscs8yvvxjzp1i9mw8s35xh";
   };
 
-  vendorHash = "sha256-Kj8gP5393++nPeX38TX6duB9OO/ql7hpRA5gTTtTl+M=";
+  vendorHash = "sha256-Y76Y3KR80dCx8+f6M0h5J6glGQGhXYE2KQM4jdJcDEM=";
 
   CGO_ENABLED = 0;
 
   ldflags = [
     "-X main.version=v${version}"
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/cluster/kubectl-gadget/default.nix:19]:
```suggestion
    "-s" "-w"
    "-X main.version=v${version}"
```

**Pull request merged**